core/src/main/scala/org/apache/spark/deploy/master/ui/MasterPage.scala

```diff
@@ -244,7 +254,12 @@ private[ui] class MasterPage(parent: MasterWebUI) extends WebUIPage("") {
     <tr>
       <td>{driver.id} {killLink}</td>
       <td>{driver.submitDate}</td>
-      <td>{driver.worker.map(w => <a href={w.webUiAddress}>{w.id.toString}</a>).getOrElse("None")}
+      <td>{driver.worker.map(w =>
+        if (parent.master.reverseProxy) {
+          <a href={"/proxy/" + w.id.toString + "/"}>{w.id.toString}</a>
```

Comment:
nit: `toString` is unnecessary.

Also, interpolation would look better: `s"/proxy/${w.id}/"`